src/sql.rs

```diff
@@ -242,6 +242,22 @@ impl Sql {
             }
         }
 
+        // Reset some options if IsChatmail is true,
+        // because there were reports from users who had these options set to something else,
+        // and then couldn't change them because they are hidden in the UI for chatmail profiles
+        if update_email_configs && context.get_config_bool(Config::IsChatmail).await? {
```

Comment:
This is also error-prone because it runs after all migrations that we might add in the future. If we in the future add a migration that simply removes `is_chatmail` from `config` table as a cleanup, then this code will do nothing and have different effect depending on whether the user upgrades release-by-release or jumps across multiple releases.